testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/faulttolerance/opentracing/OpenTracingFaultToleranceTestCase.java

```diff
@@ -0,0 +1,180 @@
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.runner.RunWith;
+
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.junit.Assert;
+import org.junit.Test;
+import org.wildfly.test.integration.microprofile.faulttolerance.opentracing.application.MockTracerFactory;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public class OpenTracingFaultToleranceTestCase {
+
+    @ArquillianResource
+    private URL url;
+
+    @Deployment(name = "opentracing-fault-tolerance")
+    public static Archive<?> deploy() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class);
+        war.addPackage(MockTracer.class.getPackage());
+        war.addPackage(MockTracerFactory.class.getPackage());
+        war.addAsServiceProvider(TracerFactory.class, MockTracerFactory.class);
+        war.addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+        return war;
+    }
+
+    @Test
+    public void testHello() throws IOException {
+        try (CloseableHttpClient client = HttpClientBuilder.create().build()) {
+            HttpResponse response = client.execute(new HttpGet(url.toExternalForm() + "rest/hello"));
+            Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+            String content = EntityUtils.toString(response.getEntity());
+            Assert.assertEquals("Hello from fallback", content);
+            response = client.execute(new HttpGet(url.toExternalForm() + "rest/tracer/spans"));
+            Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+            StringTokenizer tokenizer = new StringTokenizer(EntityUtils.toString(response.getEntity()), ";");
+            Assert.assertEquals(4, tokenizer.countTokens());
+            List<JsonObject> spans = new ArrayList<>(4);
+            while (tokenizer.hasMoreTokens()) {
+                String jsonContent = tokenizer.nextToken();
+                try (JsonReader jsonReader = Json.createReader(new StringReader(jsonContent))) {
+                    spans.add(jsonReader.readObject());
+                }
+            }
+            int parentId = 0;
+            Collections.sort(spans, (o1, o2) -> {
+                return Integer.parseInt(o1.getString("spanId")) - Integer.parseInt(o2.getString("spanId"));
+            });
+            for (int i = 0; i < spans.size(); i++) {
+                JsonObject object = spans.get(i);
+                String operationName = object.getString("operationName");
+                Assert.assertNotNull(object.toString(), object.getString("traceId"));
+                Assert.assertNotNull(object.toString(), object.getString("spanId"));
+                Assert.assertEquals(object.toString(), parentId, Integer.parseInt(object.getString("parentId")));
+                switch (i) {
+                    case 0: {
+                        Assert.assertEquals(object.toString(), operationName, "GET:org.wildfly.test.integration.microprofile.faulttolerance.opentracing.application.HelloResource.get");
```

Comment:
Please future-proof this by using `Java.lang.Class.getCanonicalName()` instead of string constants that will inevitably break with next refactoring.